### PR TITLE
use PSR-6 for caching

### PR DESCRIPTION
**Pull request by bshaffer**
addresses https://github.com/google/google-api-php-client/issues/941

### CODE REVIEW

**Comment by bshaffer** [comment]
@tmatsuo PTAL

**Comment by bshaffer** [comment]
According to the requirements of `key` in [the spec](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-6-cache.md#definitions), we will need to base64-encode our URLs to make the keys safe (i.e. we are using illegal characters in the scope urls)

**Comment by tmatsuo** [comment]
LGTM

**Comment by bshaffer** [comment]
@stanley-cheung as you implemented the original `CacheInterface`, PTAL.

File: src/CacheTrait.php [hunk 1]
```diff
@@ -62,7 +62,17 @@ private function setCachedValue($v)
...
-        $this->cache->set($key, $v);
+
+        $key = self::getValidKeyName($this->cacheConfig['prefix'].$fetcherKey);
+        $cacheItem = $this->cache->getItem($key);
+        $cacheItem->set($v);
+        $cacheItem->expiresAfter($this->cacheConfig['lifetime']);
+        return $this->cache->save($cacheItem);
+    }
+
+    public static function getValidKeyName($key)
```
Comment (kesar) [inline src/CacheTrait.php]: why is this static?

**Comment by stanley-cheung** [comment]
LGTM. Using PSR-6' `CacheItemPoolInterface` is a great idea, rather than rolling our own `CacheInterface`. Just FYI, I wasn't the original implementor of `CacheInterface`. That even predates me :P

File: src/CacheTrait.php (hunk 1 above)
Comment (bshaffer) [inline src/CacheTrait.php]: @kesar you're right, this function's visibility should match that of `getCachedValue` and `setCachedValue`

**Comment by bshaffer** [comment]
@stanley-cheung  `CacheInterface` has now been banished back to the mysterious origins from whence it came.

**Pull request merged**